### PR TITLE
Add option to retrieve comments with version history

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds an optional URL param `with_comments` to the URL that gets version history, to be used in commit code. If this URL is specified, the versions will be returned with a `comment` field.

## Testing story

- added a couple unit tests
- manual testing

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: shared/middleware/helpers/bucket_helper.rb [hunk 1]
```diff
@@ -338,19 +338,23 @@ def hard_delete_channel_content(encrypted_channel_id)
...
+  def list_versions(encrypted_channel_id, filename, with_comments: false)
     owner_id, storage_app_id = storage_decrypt_channel_id(encrypted_channel_id)
     key = s3_path owner_id, storage_app_id, filename
 
     track_list_operation 'BucketHelper.list_versions'
     s3.list_object_versions(bucket: @bucket, prefix: key).
       versions.
       map do |version|
+        comment = with_comments ?
+          DASHBOARD_DB[:project_versions].select(:comment).where(storage_app_id: storage_app_id, object_version_id: version.version_id).first&.fetch(:comment) :
```
Comment (bencodeorg) [inline shared/middleware/helpers/bucket_helper.rb]: .first guaranteed to work even when no results (ie, no safe navigation necessary)?

File: shared/middleware/files_api.rb [hunk 2]
```diff
@@ -606,7 +606,7 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
     content_type :json
 
     filename.downcase! if endpoint == 'files'
-    get_bucket_impl(endpoint).new.list_versions(encrypted_channel_id, filename).to_json
+    get_bucket_impl(endpoint).new.list_versions(encrypted_channel_id, filename, with_comments: request.GET['with_comments']).to_json
```
Comment (bencodeorg) [inline shared/middleware/files_api.rb:609]: Not a comment for your work, but what is the deal with needing all caps GET to access a param? Seems odd to me

File: shared/middleware/helpers/bucket_helper.rb (hunk 1 above)
Comment (bencodeorg) [reply shared/middleware/helpers/bucket_helper.rb]: And for my understanding, this is sequel? Would be nice to split this onto multiple lines if possible, but maybe causes issues in a ternary?

**Review by bencodeorg — APPROVED** [review]

File: shared/middleware/files_api.rb (hunk 2 above)
Comment (bethanyaconnor) [reply shared/middleware/files_api.rb:609]: I have no idea, especially as it seems to work without the `.GET`. I'm going to keep this as is to be inline with the rest of the file, unless you object.

File: shared/middleware/helpers/bucket_helper.rb (hunk 1 above)
Comment (bethanyaconnor) [reply shared/middleware/helpers/bucket_helper.rb]: > .first guaranteed to work even when no results (ie, no safe navigation necessary)?

Yes, I believe `where` always returns an array so `first` will work, but might itself return `nil`. Below is an excerpt from calling this in my browser. Some versions have a comment, some don't.
```{"versionId":"xoIslfy9nAZuXxpn.eQrcC8.g.O387KN","lastModified":"2021-08-12T16:33:11.000Z","comment":"hi","isLatest":false},{"versionId":"es3D.s0lw3r3oH5FdM.fwQ88cLzUGos5","lastModified":"2021-08-12T16:30:42.000Z","isLatest":false},{"versionId":"TbMd14CRSoJ6iw8hIbWt7HmTF.AKJOds","lastModified":"2021-08-12T16:25:49.000Z","isLatest":false},```

File: shared/middleware/files_api.rb (hunk 2 above)
Comment (bencodeorg) [reply shared/middleware/files_api.rb:609]: haha weird! yeah totally fine, was just observing

File: shared/middleware/helpers/bucket_helper.rb (hunk 1 above)
Comment (bethanyaconnor) [reply shared/middleware/helpers/bucket_helper.rb]: I can break up the line (sorry got distracted)

**Pull request merged**